pontoon/base/views.py

```diff
@@ -708,59 +708,34 @@ def translation_memory(request):
     max_dist = math.floor(min(length / min_quality, 1000))
 
     # Only check entities with similar length
-    entities = Entity.objects.all().extra(
-        where=["CHAR_LENGTH(string) BETWEEN %s AND %s" % (min_dist, max_dist)])
+    entries = TranslationMemoryEntry.objects.filter(locale=locale).extra(
+        where=['(CHAR_LENGTH(source) BETWEEN %s AND %s)'\
+              'AND levenshtein_ratio(source, %s) > %s'],
+        params=(min_dist, max_dist, text, min_quality),
+        select={'quality': 'levenshtein_ratio(source, %s) * 100'},
+        select_params=(text,)).exclude(entity__pk=int(pk))
```

Comment:
@mathjazz fixed! :)

On Mon, Dec 21, 2015 at 2:19 PM, Jarek Śmiejczak <jot@smiejczak.net> wrote:

> @mathjazz comma is required, because otherwise it will use whole string as
> sequence and that will generate brokern query.
>
> On Mon, Dec 21, 2015 at 12:31 PM, Matjaž Horvat <notifications@github.com>
> wrote:
>
>> In pontoon/base/views.py
>> <https://github.com/mozilla/pontoon/pull/286#discussion_r48137668>:
>>
>> > @@ -708,59 +708,34 @@ def translation_memory(request):
>> >      max_dist = math.floor(min(length / min_quality, 1000))
>> >
>> >      # Only check entities with similar length
>> > -    entities = Entity.objects.all().extra(
>> > -        where=["CHAR_LENGTH(string) BETWEEN %s AND %s" % (min_dist, max_dist)])
>> > +    entries = TranslationMemoryEntry.objects.filter(locale=locale).extra(
>> > +        where=['(CHAR_LENGTH(source) BETWEEN %s AND %s)'\
>> > +              'AND levenshtein_ratio(source, %s) > %s'],
>> > +        params=(min_dist, max_dist, text, min_quality),
>> > +        select={'quality': 'levenshtein_ratio(source, %s) * 100'},
>> > +        select_params=(text,)).exclude(entity__pk=int(pk))
>>
>>
>>    1. That exclude with break in /terminology or Search tab, because pk
>>    is not set.
>>    2. No need for comma in text,.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/mozilla/pontoon/pull/286/files#r48137668>.
>>
>
>
>
> --
> Kind regards
>
> ​Jarek "jotes" Śmiejczak,
> ​P​
> ython programmer​ by day, mozillian by heart..
> Homepage: http://joyfulhack.in, Github: http://github.com/jotes
> Mobile: +48693027040, Mozillian: http://mozillians.org/u/jotes
>
>


-- 
Kind regards

​Jarek "jotes" Śmiejczak,
​P​
ython programmer​ by day, mozillian by heart..
Homepage: http://joyfulhack.in, Github: http://github.com/jotes
Mobile: +48693027040, Mozillian: http://mozillians.org/u/jotes